kratos/elements/distance_calculation_flux_based_element.h

```diff
@@ -0,0 +1,359 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ \.
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//			 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Daniel Diez, Pablo Becker
+//
+
+#if !defined(KRATOS_DISTANCE_CALCULATION_FLUX_BASED_ELEMENT)
+#define  KRATOS_DISTANCE_CALCULATION_FLUX_BASED_ELEMENT
+
+// System includes
+
+// Project includes
+#include "includes/define.h"
+#include "includes/element.h"
+#include "includes/variables.h"
+#include "includes/serializer.h"
+#include "utilities/geometry_utilities.h"
+#include "geometries/geometry_data.h"
+
+namespace Kratos
+{
+
+/*The "DistanceCalculationFluxBasedElement" element formulation is meant to compute the pseudo filltime or flowlength for a filling problem
+Currently only supports very viscous problems (polymers or RTM)
+It requires two steps to be performed:
+1. Solves a transient diffusion problem to get a field equivalent to a potential.
+2. from the field, compute gradients to obtain velocity and use it to compute flowlenght/filltime.
+*/
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+template< unsigned int TDim,  unsigned int TNumNodes >
+class KRATOS_API(KRATOS_CORE) DistanceCalculationFluxBasedElement : public Element
+{
+public:
+
+    /// Counted pointer of
+    KRATOS_CLASS_INTRUSIVE_POINTER_DEFINITION(DistanceCalculationFluxBasedElement);
+
+    ///@name Type Definitions
+    ///@{
+
+    typedef Node<3> NodeType;
+    typedef Geometry<NodeType> GeometryType;
+    typedef Geometry<NodeType>::PointsArrayType NodesArrayType;
+    typedef Vector VectorType;
+    typedef Matrix MatrixType;
+    typedef std::size_t IndexType;
+    typedef std::size_t SizeType;
+    typedef std::vector<std::size_t> EquationIdVectorType;
+    typedef std::vector< Dof<double>::Pointer > DofsVectorType;
+    typedef PointerVectorSet<Dof<double>, IndexedObject> DofsArrayType;
+    typedef GeometryType::ShapeFunctionsGradientsType ShapeFunctionsGradientsType;
```

Comment:
Instead of creating new type definitions the base class one must be reused as much as possible. This eases the maintenance of the code in the case of an eventual API change of the base `Element` class.

Besides I think that we should promote the `using` syntax instead of the `typedef` one considering that this is a new file addition. Thought in this case it makes no difference, it is a good practice for the future IMO (see a discussion [here](https://stackoverflow.com/questions/10747810/what-is-the-difference-between-typedef-and-using-in-c11)).

Reply:
ok